### ISSUE TITLE
feat: Camera Plugin events

### ISSUE THREAD

**Issue by kedarguy** [ionic-team/capacitor#2802]
# Feature Request

## Describe the Feature Request
Camera plugin events -

- modal opened
- modal closed without choosing photo

## Platform Support Requested

- [x] Android
- [x] iOS
- [ ] Electron
- [ ] Web

## Describe Preferred Solution
Add events to the Camera plugin so it is possible to know when the choose camera/gallery modal opens and if it closed without choosing a photo

## Related Code

I think we can use the Camera.addListener api, and add the following events:

"modal opened"
"modal closed"
possibly more events for other scenarios where a photo wasn't chosen

## Additional Context

When using the Plugin, sometimes the modal wont open immediately (in my case it is because another capacitor Plugin is working in the background) 

The problem is that I cant know when to disable and re-enable the button to get a new photo, so the user might invoke the Camera.getPhoto function a few times, which then causes the modal to pop a few times after choosing a photo.

**Comment by jcesarmobile** [comment]
Issues tagged with feature request are closed but tracked for👍  reactions to gauge interest.

BTW, if you cancel the photo the promise is rejected with an error, that should be enough to know if the user canceled it

**Comment by kedarguy** [comment]
> Issues tagged with feature request are closed but tracked for👍 reactions to gauge interest.
> 
> BTW, if you cancel the photo the promise is rejected with an error, that should be enough to know if the user canceled it
 
@jcesarmobile Thanks for the response.

I checked again on ios I do get a rejection with a correct error message: "User cancelled photos app"

 
After further testing on android I found out that if you choose the photos or camera options you do get a rejection, but if you cancel before choosing one of the options, there is no rejection

should edit the issue as a bug? or open an new issue?
